src/cpp/rtps/reader/StatelessReader.cpp

```diff
@@ -115,13 +120,15 @@ bool StatelessReader::matched_writer_add(
             return false;
         }
 
-        if (m_att.durabilityKind != VOLATILE)
+        // Intraprocess manages durability itself
+        if (!is_same_process && m_att.durabilityKind != VOLATILE)
         {
             // simulate a notification to force reading of transient changes
             datasharing_listener_->notify(false);
         }
     }
-    else
+
+    if (!is_datasharing || is_same_process)
     {
         RemoteWriterInfo_t info;
         info.guid = wdata.guid();
```

Comment:
I think this code (except the one adding the writer to `matched_writers_`) should always be executed. Otherwise, we will not add persistence guid mapping information for remote data-sharing writers.

Looking carefully at this, liveliness management could also be affected.

It may be worth adding a `is_datasharing` flag to `RemoteWriterInfo_t`, and always add the writer to `matched_writers_`